.github/workflows/run-matrix.yml

```diff
@@ -38,3 +38,4 @@ jobs:
         with:
           name: coverage-reports
           path: "**/.coverage*"
+          include-hidden-files: true
```

Comment:
It's such a coincidence that I've been reading about this default behaviour recently.

Reply:
If possible, can we add `- uses: actions/checkout@v4` with `persist-credentials: false`:

> The auth token is persisted in the local git config. This enables your scripts to run authenticated git commands. The token is removed during post-job cleanup. Set persist-credentials: false to opt-out.

https://github.com/actions/checkout/tree/v4/